decoder.go

```diff
@@ -199,6 +194,27 @@ func (d *decoder) setFieldByType(current reflect.Value, namespace []byte, idx in
 			}
 		}
 	}
+	{
```

Comment:
I'm not sure the extra scope is needed.

Reply:
This was added so that v could be shadowed and modified without changing the code later on, but I'll just assign all modifications to a new var instead.